go/vt/tabletserver/query_executor.go

```diff
@@ -295,25 +295,33 @@ func (qre *QueryExecutor) execDDL() (*sqltypes.Result, error) {
 		return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "DDL is not understood")
 	}
 
-	conn, err := qre.tsv.te.txPool.LocalBegin(qre.ctx)
+	if qre.transactionID != 0 {
+		conn, err := qre.tsv.te.txPool.Get(qre.transactionID, "for query")
+		if err != nil {
+			return nil, err
+		}
+		defer conn.Recycle()
+		result, err := qre.execSQL(conn, qre.query, false)
+		if err != nil {
+			return nil, err
+		}
+		conn.ReloadSchemaOnCommit = true
+		return result, nil
+	}
+
+	result, err := qre.execAsTransaction(func(conn *TxConnection) (*sqltypes.Result, error) {
```

Comment:
Don't know why the other code was hanging, but this is the right thing to do anyway.